packages-ts/starknet-gauntlet/src/commands/base/executeCommand.ts

```diff
@@ -124,27 +129,93 @@ export const makeExecuteCommand = <UI, CI>(config: ExecuteCommandConfig<UI, CI>)
         contract: new Contract(c.contract.abi, c.contractAddress ?? '', c.provider.provider),
       }
 
+      const overrideExecutionContext = (contractAddress: string): ExecutionContext => {
+        return {
+          ...c.executionContext,
+          contract: new Contract(c.contract.abi, contractAddress, c.provider.provider),
+          contractAddress,
+        }
+      }
+
       const rdd = flags.rdd || process.env.RDD
       if (rdd) {
         deps.logger.info(`Using RDD from ${rdd}`)
         c.executionContext.rdd = getRDD(rdd)
+
+        // If more than one argument / address is passed to the command, then we'll assume that we
+        // want to perform a batch operation where we execute multiple calls from one transaction.
+        // If the RDD flag is also specified, then we gain the flexibility to vary the inputs of each
+        // call. Instead of forwarding all args to `buildCommandInput`, we can pass each arg to the
+        // function individually:
+        //
+        //   makeBuildInput(..., [args[0]], ..., <context with contract object and contract address replaced with args[0]>)
+        //   makeBuildInput(..., [args[1]], ..., <context with contract object and contract address replaced with args[1]>)
+        //   makeBuildInput(..., [args[2]], ..., <context with contract object and contract address replaced with args[2]>)
+        //   ...
+        //
+        // Under the hood, each call will access the RDD JSON config file at key `args[i]` which will
+        // return a different set of config values. These values can be used to populate the transaction
+        // calls.
+        //
+        if (args.length > 1) {
+          c.batchInput = new Map(
+            await Promise.all(
+              args.map(async (addr: string) => [
+                addr,
+                await c.buildCommandInput(flags, [addr], env, overrideExecutionContext(addr)),
+              ]),
+            ),
+          )
+        }
       }
 
       c.input = await c.buildCommandInput(flags, args, env)
 
-      c.beforeExecute = config.hooks?.beforeExecute
-        ? config.hooks.beforeExecute(c.executionContext, c.input, {
-            logger: deps.logger,
-            prompt: deps.prompt,
+      if (c.batchInput != null) {
```

Comment:
wondering, to clean up this logic can we just always assume we have a batch? even if it's of size 1?

Reply:
Yup good point - I'll clean this up in a moment